R/summary_functions.R

```diff
@@ -4,8 +4,9 @@
 #' @export
 percent_coverage <- function(coverage_result){
   coverage_data <- as.data.frame(coverage_result)
-  num_lines <- nrow(coverage_data)
-  num_not_zero <- sum(coverage_data$value != 0) # how many lines have a value that is not zero?
+  num_lines <- length(unique(coverage_data$first_line))
```

Comment:
I believe the https://github.com/MangoTheCat/testCoverage package
highlights only the portion of the line that includes the expression.  The
srcref contains the columns a given expression covers in the source file,
so it is possible to do.

coveralls only reports coverage per line, so it does make sense to have a
parameter, or different functions (line_coverage, expression_coverage
perhaps) to calculate each percentage.

On Tue, Jan 13, 2015 at 10:25 AM, Karl Forner <notifications@github.com>
wrote:

> In R/summary_functions.R
> <https://github.com/jimhester/covr/pull/30#discussion-diff-22868597>:
>
> > @@ -4,8 +4,9 @@
> >  #' @export
> >  percent_coverage <- function(coverage_result){
> >    coverage_data <- as.data.frame(coverage_result)
> > -  num_lines <- nrow(coverage_data)
> > -  num_not_zero <- sum(coverage_data$value != 0) # how many lines have a value that is not zero?
> > +  num_lines <- length(unique(coverage_data$first_line))
>
> Anyway we could provide, via a parameter or different functions the two
> behaviors.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jimhester/covr/pull/30/files#r22868597>.
>